lib/Index/SymbolIndex.cpp

```diff
@@ -500,6 +503,33 @@ bool SymbolIndexImpl::foreachSymbolInFilePath(CanonicalFilePathRef filePath,
     return didFinish;
 }
 
+bool SymbolIndexImpl::foreachSymbolOccurrenceInFilePath(CanonicalFilePathRef filePath,
+                                                        function_ref<bool(SymbolOccurrenceRef Occur)> Receiver) {
+    bool didFinish = true;
+    ReadTransaction reader(DBase);
+
+    IDCode filePathCode = reader.getFilePathCode(filePath);
+    reader.foreachUnitContainingFile(filePathCode, [&](ArrayRef<IDCode> idCodes) -> bool {
+        for (IDCode idCode : idCodes) {
+            UnitInfo unitInfo = reader.getUnitInfo(idCode);
+
+            for (UnitInfo::Provider provider : unitInfo.ProviderDepends) {
+                IDCode providerCode = provider.ProviderCode;
+                if (provider.FileCode == filePathCode) {
+                    auto record = createVisibleProviderForCode(providerCode, reader);
+                    didFinish = record->foreachSymbolOccurrence(Receiver);
+
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    });
```

Comment:
Could you change this to 2-space indentation?

Reply:
done